logstash-core/src/main/java/org/logstash/execution/Filter.java

```diff
@@ -0,0 +1,69 @@
+package org.logstash.execution;
+
+import org.logstash.Event;
+
+/**
+ * A Filter is simply a mapping of {@link QueueReader} to a new {@link QueueReader}.
+ */
+public interface Filter extends AutoCloseable {
+
+    QueueReader filter(QueueReader reader);
+
+    void flush(boolean isShutdown);
+
+    @LogstashPlugin(name = "mutate")
+    final class Mutate implements Filter {
+
+        private final String field;
+
+        private final String value;
+
+        /**
+         * Required Constructor Signature only taking a {@link LsConfiguration}.
+         * @param configuration Logstash Configuration
+         * @param context Logstash Context
+         */
+        public Mutate(final LsConfiguration configuration, final LsContext context) {
+            this.field = configuration.getString("ls.plugin.mutate.field");
+            this.value = configuration.getString("ls.plugin.mutate.value");
```

Comment:
Why do we need the boilerplate of `ls.plugin.mutate.*`, why can't we just have `configuration.getString("value")`. 

Can you go into a bit more detail as to how `LsConfiguration` would work? Is there a way to declare: "field must be a string" as you can today? This is important for driving a visual builder in the future.

Reply:
@andrewvc you're right the prefixes aren't necessary :) WIll remove them :)

Reply:
@andrewvc I'd suggest that a plugin has to implement a method that returns `Map<String, LsConfigTypeInfo> configSchema()`  that describes each key's expected value (type, deprecated etc.).
Then we can use the return of that for a visual builder + also write validation logic using this map into the `ConfigCompiler` :) I'll add some code illustration for that approach soon :)

Reply:
I think that's fine for the scope here.

In the future we should improve this to drive the visual builder. It'd be great to support a nested description as well. I'd love @ycombinator 's input here. What do we need to drive a pipeline builder UI? We should create a new issue for that discussion however. I think parity with the existing API is good for v1 here.

Reply:
In a perfect world (likely not v1) the interface of a plugin (properties, their data types, any additional validation rules, short descriptive documentation, etc.) could be described in a language-agnostic file. That file could then be used (perhaps after further "compilation") by LS code as well as the LS UI. I'm deliberately leaving out the specifics of "used".

Alternatively, such an interface definition file could be generated from one of these classes as well, if that's possible.

The crux of the problem I'm trying to solve her is sharing of definitions between core and UI so we don't have to duplicate them in both places, keep them in sync, etc.

Of course, we shouldn't let this concern weigh down v1 unless solving for it later from the v1 design is going to be difficult.